### PR TITLE
make keyring optional

### PR DESCRIPTION
**Pull request by meeuw**
## Description
make keyring optional

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: pgcli/main.py [hunk 1]
```diff
@@ -440,11 +461,22 @@ def connect(self, database='', host='', user='', port='', passwd='',
             try:
                 pgexecute = PGExecute(database, user, passwd, host, port, dsn,
                                       application_name='pgcli', **kwargs)
-                if passwd:
+                if passwd and keyring and self.config["main"].get("keyring", True):
                     try:
                         keyring.set_password('pgcli', key, passwd)
-                    except (keyring.errors.InitError, RuntimeError):
-                        pass
+                    except (
```
Comment (j-bennet) [inline pgcli/main.py]: This keyring-related handling is getting long enough to be extracted into methods, or perhaps a class that `main` can delegate things to. Our `main` is very clattered and it's becoming difficult to follow the logic.

File: pgcli/main.py [hunk 2]
```diff
@@ -440,11 +461,22 @@ def connect(self, database='', host='', user='', port='', passwd='',
             try:
                 pgexecute = PGExecute(database, user, passwd, host, port, dsn,
                                       application_name='pgcli', **kwargs)
-                if passwd:
+                if passwd and keyring and self.config["main"].get("keyring", True):
```
Comment (j-bennet) [inline pgcli/main.py]: I would add the default `keyring = True` to `pgclirc`, for better discoverability, and I'd also put in a comment explaining what it is. People tend to rely on `pgclirc` to tell them what `pgcli` can do.

Comment (meeuw) [reply pgcli/main.py]: Ah yes, thanks for reminding. I forgot to change pgclirc

File: pgcli/main.py (hunk 1 above)
Comment (meeuw) [reply pgcli/main.py]: Hmm yes, but I don't like to generalize a generalization (which I think keyring is). You're right about it's getting long and `main.py` is flooded with exceptions like this. But I don't know how that can be fixed.

I've moved `keyring.set_password` so it only has to be called once.

**Comment by j-bennet** [comment]
Merging! Release coming up.

**Pull request merged**